### ISSUE TITLE
Hits to external firewall

### ISSUE THREAD

**Issue by jamesoxford08** [webmin/webmin#347]
I've used webmin for a long time which makes writing this difficult.  I've noticed this for the last 2-3 months probably, and it wasn't until I saw the mirrors change and the hash mismatches that it made me worry/wonder.  After installing webmin, if I open a port on that machine, I'll start getting firewall hits at the external gateway from random addresses to the newly opened port number. I'm really at a loss for how this is happening though unless a system has access to the internal network.  There are no network services open.  There is a bind server on the webmin server that does recursive dns queries to authoritative nameservers only.  The server forwards all requests for the internal domain to another server and is the only forwarder set up for the internal domain dns.  Other than that and repository access, it should have no internet contact; but sure enough; I open 3306; I get hits for 3306 on the firewall.  I change the default port for webmin to a random ephemeral port, hits to the firewall for that port number.  Anyone experiencing something similar? Or have an explanation that might explain this?

**Comment by jcameron** [comment]
I don't think this is related to the hash mismatches! That was caused by a mirror being out of sync..

Rather, there is probably just someone doing port scanning on ALL ports.

**Comment by jamesoxford08** [comment]
This is the same jcameron of jcameron-key.asc fame? I wasn't saying it was because of the hash mismatch; or directly.  I tend to pick up coincidence and patterns pretty quickly; but not all coincidence is a pattern; sometimes it really is just coincidence.  The somersettech mirror has been up as long as I've been using Webmin.  The logs I have definitely isn't just port knocking wide swaths of ports; and I can explain away more targeted port scans on common ports, ie 21:22,80,8080.  And this hasn't been a one day thing; I look at the external log almost daily and it's set up to log all dropped packets.  I just installed a mysql server in the last few months and it had been 2-3 years since I'd had one up before that.  I don't remember seeing hits to 3306 before I installed one most recently, and everything went in at the same time; fresh VM, new OS, new webmin installation; and I remember that night I had a bunch of hits to 3306 and at the time I just thought it was kind of a weird time to start seeing hits on 3306.  3306 I can attribute to portscanning.  Its 10000 and then when I change it to a port in the 47000 range and see a hit to that specific port, and no ports around it, that I have trouble explaining.  

The only thing the hash made me worry about was whether or not the creators/owners of webmin were still in control of the domain and the repository.
